hikari/webhooks.py

```diff
@@ -761,71 +798,164 @@ async def fetch_self(self, *, use_token: undefined.UndefinedOr[bool] = undefined
+            however, Discord implements other undocumented rules for
+            rate-limiting, such as limits per attribute. These cannot be
+            detected or handled normally by Hikari due to their undocumented
+            nature, and will trigger this exception if they occur.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """  # noqa: E501 - Line too long
+        webhook = await self.app.rest.edit_webhook(
+            self.id,
+            name=name,
+            avatar=avatar,
+            channel=channel,
+            reason=reason,
+        )
+        assert isinstance(webhook, ChannelFollowerWebhook)
+        return webhook
+
+    async def fetch_self(self) -> ChannelFollowerWebhook:
+        """Fetch this webhook.
 
         Returns
         -------
-        typing.Optional[hikari.files.URL]
-            The URL of the resource. `builtins.None` if no avatar is set (in
-            this case, use the `default_avatar` instead).
+        hikari.webhooks.ChannelFollowerWebhook
+            The requested webhook object.
 
         Raises
         ------
-        builtins.ValueError
-            If `size` is not a power of two between 16 and 4096 (inclusive).
+        hikari.errors.ForbiddenError
+            If you're not in the guild that owns this webhook or
+            lack the `MANAGE_WEBHOOKS` permission.
+        hikari.errors.NotFoundError
+            If the webhook is not found.
+        hikari.errors.UnauthorizedError
+            If you pass a token that's invalid for the target webhook.
+        hikari.errors.RateLimitTooLongError
+            Raised in the event that a rate limit occurs that is
+            longer than `max_rate_limit` when making a request.
+        hikari.errors.RateLimitedError
+            Usually, Hikari will handle and retry on hitting
+            rate-limits automatically. This includes most bucket-specific
+            rate-limits and global rate-limits. In some rare edge cases,
+            however, Discord implements other undocumented rules for
+            rate-limiting, such as limits per attribute. These cannot be
+            detected or handled normally by Hikari due to their undocumented
+            nature, and will trigger this exception if they occur.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
         """
-        if self.avatar_hash is None:
-            return None
+        webhook = await self.app.rest.fetch_webhook(self.id)
+        assert isinstance(webhook, ChannelFollowerWebhook)
+        return webhook
 
-        return routes.CDN_USER_AVATAR.compile_to_file(
-            urls.CDN_URL,
-            user_id=self.id,
-            hash=self.avatar_hash,
-            size=size,
-            file_format=ext,
-        )
+
+@attr.define(hash=True, kw_only=True, weakref_slot=False)
```

Comment:
```suggestion
@attr.define(hash=True, kw_only=True, weakref_slot=False, auto_attribs=False)
```

Reply:
Don't see why that'd need to be disabled, doesn't really change anything here

Reply:
Stops attr from adding attributes to it, doesnt hurt to add

Reply:
Gonna keep redefining the attr field since this ensures field.type is correct for this field